### ISSUE TITLE
since_commit is broken

### ISSUE THREAD

**Issue by dmrickert** [dxa4481/truffleHog#218]
The since_commit feature is broken.  

It always scans the first commit of a repo due to the way the for loop "continue"s to loop through all commits until it reaches the oldest commit.  Further, the since_commit feature actually ignores the commit newer then the since_commit due to the way the prev_commit logic is written.

**Comment by dmrickert** [comment]
To expand a bit, let's take a look at the tests and the Git log for the project the tests rely on.

```
commit 9ed54617547cfca783e0f81f8dc5c927e3d1e345
Author: flower <flower@flowers-MacBook-Pro.local>
Date:   Fri Dec 30 23:20:50 2016 -0600

    OH no a secret

commit 709f22821820a7815106f82a03e8d90f50b2b653
Author: flower <flower@flowers-MacBook-Pro.local>
Date:   Fri Dec 30 23:19:51 2016 -0600

    removing secret

commit d15627104d07846ac2914a976e8e347a663bbd9b
Author: flower <flower@flowers-MacBook-Pro.local>
Date:   Fri Dec 30 23:19:39 2016 -0600

    Oh no a secret file

commit 202564cf776b402800a4aab8bb14fa4624888475
Author: flower <flower@flowers-MacBook-Pro.local>
Date:   Fri Dec 30 23:16:57 2016 -0600

    updating the readme
```

So, a secret was added in d156... removed in 709f... and then added again in 9ed5.  The tests were set up to use d156 as the "since_commit".  This should've caught the secret being removed in 709f.  Instead, the since_commit actually caught the change in 9ed.  This was because of the bug I mentioned where the first commit since the designated "since_commit" actually gets ignored.

Also, the first commit was always getting scanned when the since_commit flag was used.  If you look at the logic in truffleHog.py specifically in the function "find_strings" the behavior once "since_commit" is matched is to keep looping through `repo.iter_commits(branch_name, max_count=max_depth)`.  While it keeps looping through, it continues to set `prev_commit = curr_commit` before it hits the continue.  Also, curr_commit keeps getting updated as it loops through the for loop and hits the continue as well.

That final curr_commit ends up always hitting the logic where it compares the commit to a "NULL_TREE" which is a blank git construct.  Thus, the first commit on a branch was always getting scanned when using since_commit.

Love the tool, impressed that it's so powerful with such few lines of code.

**Comment by toshith** [comment]
Hi Dough....How did you fix the since_commit parameter.??? I just wanted to scan the repository which is last committed to github using trufflehog...Your help will be higly appreciated.

**Comment by dxa4481** [comment]
This should be fixed with #219 feel free to re-open if something's wonky.

Note though, the branch flag might be needed if there's a commit in one branch but not another.
